File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        res = 0
        bCount = 0

        for i in s:
            if i == "b":
                bCount += 1
            else:
                res = min(res + 1, bCount)
        return res

        

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int d = 0;
        int countB = 0;

        for (int c=0;c<s.length();c++) {
            if (s.charAt(c) == 'b') {
                countB++;
            } else {
                d = Math.min(d + 1, countB);
            }
        }

        return d;
    }
}
